benchmark/url/urlpattern-test.js

```diff
@@ -0,0 +1,29 @@
+'use strict';
+const common = require('../common.js');
+const { URLPattern } = require('url');
+const { notStrictEqual } = require('assert');
+
+const tests = [
+  'https://(sub.)?example(.com/)foo',
+  { 'hostname': 'xn--caf-dma.com' },
+  { 'pathname': '/foo', 'search': 'bar', 'hash': 'baz',
+    'baseURL': 'https://example.com:8080' },
+  { 'pathname': '/([[a-z]--a])' },
+];
+
+const bench = common.createBenchmark(main, {
+  pattern: tests.map(JSON.stringify),
+  n: [1e5],
+});
+
+function main({ pattern, n }) {
+  const inputPattern = JSON.parse(pattern);
+  const urlpattern = new URLPattern(inputPattern);
+
+  bench.start();
+  for (let i = 0; i < n; i += 1) {
+    const p = urlpattern.test('https://sub.example.com/foo');
+    notStrictEqual(p, undefined);  // We don't care if it is true or false.
+  }
+  bench.end(n);
```

Comment:
```suggestion
  let deadCode = undefined;
  bench.start();
  for (let i = 0; i < n; i += 1) {
    deadCode = urlpattern.test('https://sub.example.com/foo');
  }
  bench.end(n);
  notStrictEqual(deadCode, undefined);  // We don't care if it is true or false.
```

You can leave the `ok` out to avoid perf issues related to the assert ok.